src/Twig/AverageColorFromImageFilterExtension.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace App\Twig;
+
+use Twig\Extension\AbstractExtension;
+use Twig\TwigFilter;
+
+/**
+ * Description of AgoFilterExtension.
+ */
+class AverageColorFromImageFilterExtension extends AbstractExtension
+{
+    public function getFilters(): array
+    {
+        return [
+            new TwigFilter('average_color', $this->getAverageColorFromImage(...)),
+        ];
+    }
+
+    public function getAverageColorFromImage($imagePath): string
+    {
+
+        $image = $this->createImageFromFile($imagePath);
+
+        if($image === false) {
+            return '#000000';
+        }
+
+        $pixIm = imagescale($image, 1,1);
+        $rgb = imagecolorsforindex($pixIm, imagecolorat($pixIm, 0, 0));
+
+        return sprintf("#%02x%02x%02x", $rgb["red"], $rgb["green"], $rgb["blue"]);
+    }
+
+    private function createImageFromFile(string $imagePath): \GdImage|false {
```

Comment:
`use` statement for GdImage
maybe use `null` instead of false for "no image"

Reply:
false is returned from imagecreatefrom*** if an error happened. So false to be complient with that.